tck/features/TernaryLogicAcceptance.feature

```diff
@@ -0,0 +1,160 @@
+      """
+      RETURN null IS NULL AS value
+      """
+    Then the result should be:
+      | value |
+      | true  |
+    And no side effects
+
+  Scenario: A literal null is not IS NOT NULL
+    When executing query:
+      """
+      RETURN null IS NOT NULL AS value
+      """
+    Then the result should be:
+      | value |
+      | false |
+    And no side effects
+
+  Scenario: It is unknown - i.e. null - if a null is equal to a null
+    When executing query:
+      """
+      RETURN null = null AS value
+      """
+    Then the result should be:
+      | value |
+      | null  |
+    And no side effects
+
+  Scenario: It is unknown - i.e. null - if a null is not equal to a null
+    When executing query:
+      """
+      RETURN null <> null AS value
+      """
+    Then the result should be:
+      | value |
+      | null  |
+    And no side effects
+
+  Scenario Outline: Using null in AND
+    And parameters are:
+      | lhs    | <lhs>    |
+      | rhs    | <rhs>    |
+      | result | <result> |
+    When executing query:
+      """
+      RETURN {lhs} AND {rhs} AS result
+      """
+    Then the result should be:
+      | result   |
+      | <result> |
+    And no side effects
+
+    Examples:
+      | lhs   | rhs   | result |
+      | null  | null  | null   |
+      | null  | true  | null   |
+      | true  | null  | null   |
+      | null  | false | false  |
+      | false | null  | false  |
+
+  Scenario Outline: Using null in OR
+    And parameters are:
+      | lhs    | <lhs>    |
+      | rhs    | <rhs>    |
+      | result | <result> |
+    When executing query:
+      """
+      RETURN {lhs} OR {rhs} AS result
+      """
+    Then the result should be:
+      | result   |
+      | <result> |
+    And no side effects
+
+    Examples:
+      | lhs   | rhs   | result |
+      | null  | null  | null   |
+      | null  | true  | true   |
+      | true  | null  | true   |
+      | null  | false | null   |
+      | false | null  | null   |
+
+  Scenario Outline: Using null in XOR
+    And parameters are:
+      | lhs    | <lhs>    |
+      | rhs    | <rhs>    |
+      | result | <result> |
+    When executing query:
+      """
+      RETURN {lhs} XOR {rhs} AS result
+      """
+    Then the result should be:
+      | result   |
+      | <result> |
+    And no side effects
+
+    Examples:
+      | lhs   | rhs   | result |
+      | null  | null  | null   |
+      | null  | true  | null   |
+      | true  | null  | null   |
+      | null  | false | null   |
+      | false | null  | null   |
+
+  Scenario Outline: Using null in IN
+    And parameters are:
+      | elt    | <elt>    |
+      | coll   | <coll>   |
+      | result | <result> |
+    When executing query:
+      """
+      RETURN {elt} IN {coll} AS result
+      """
+    Then the result should be:
+      | result   |
+      | <result> |
+    And no side effects
+
+    Examples:
+      | elt  | coll            | result |
+      | null | null            | false  |
```

Comment:
Let's do that once we have that annotation and keep it as is for now - I've cardified accordingly